packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -87,7 +87,7 @@ For example, to filter locales on only germany \`localeFilter: locale => locale.
 
 List of locales and their codes can be found in Contentful app -> Settings -> Locales`
         )
-        .default(() => true),
+        .default(() => undefined),
```

Comment:
```suggestion
        .default(() => () => true),
```

As pointed out in https://github.com/gatsbyjs/gatsby/pull/27803 the default should probably be `() => true`